cdap-ui/app/services/callback-helper.js

```diff
@@ -0,0 +1,34 @@
+angular.module(PKG.name + '.services')
+  .service('MyCallbackHelper', function($timeout) {
+    var DEFAULT_DELAY = 500;   // default delay in milliseconds between clicks
+
+    /**
+     * Returns a function that when called, will only call the {@param callback} if the given delay has passed.
+     */
+    this.doWithDelay = function(callback, delay) {
+      var disabled = false;
+      if (typeof delay === 'undefined') {
+        delay = DEFAULT_DELAY;
+      }
+      return function() {
+        if (!disabled) {
+          disabled = true;
+          $timeout(function () { disabled = false; }, delay, false);
+          callback.apply(this, arguments);
```

Comment:
Sorry, maybe I should rename this. It's more like 'call the callback if it has been X milliseconds f delay since the last call'.